pallets/watchtower/src/lib.rs

```diff
@@ -809,6 +1028,142 @@ pub mod pallet {
+                challenge_count: 1,
+                trigger: ChallengeAdminTrigger::ConsensusReached,
+            });
+
+            VoteCounters::<T>::remove(summary_instance, &root_id);
+            let _ = VoterHistory::<T>::clear_prefix(&consensus_key, u32::MAX, None);
+            VotingStartBlock::<T>::remove(&consensus_key);
+            PendingValidationRootHash::<T>::remove(&consensus_key);
+            ConsensusThreshold::<T>::remove(&consensus_key);
+
+            Ok(())
+        }
+
+        fn handle_voting_period_expired(
+            summary_instance: SummarySource,
+            root_id: RootId<BlockNumberFor<T>>,
+        ) -> DispatchResult {
+            let (yes_votes, no_votes) = VoteCounters::<T>::get(summary_instance, root_id.clone());
+
+            if no_votes > yes_votes {
+                Self::create_automatic_challenge(summary_instance, root_id)
+            } else {
+                Self::handle_expired_voting_without_challenges(summary_instance, root_id)
+            }
+        }
+
+        fn calculate_consensus_threshold(total_watchtowers: u32) -> u32 {
+            (total_watchtowers * 2) / 3
+        }
+
+        fn check_consensus_reached(
+            instance: SummarySource,
+            root_id: &RootId<BlockNumberFor<T>>,
+        ) -> Result<(bool, u32, u32), Error<T>> {
+            let consensus_key = (instance, root_id.clone());
+            let (positive_votes, _) = VoteCounters::<T>::get(instance, root_id.clone());
+
+            let required_for_consensus =
+                ConsensusThreshold::<T>::get(&consensus_key).ok_or(Error::<T>::VotingNotStarted)?;
+            let consensus_reached = positive_votes >= required_for_consensus;
+            Ok((consensus_reached, positive_votes, required_for_consensus))
+        }
+
+        fn handle_expired_voting_without_challenges(
+            instance: SummarySource,
+            root_id: RootId<BlockNumberFor<T>>,
+        ) -> DispatchResult {
+            let consensus_key = (instance, root_id.clone());
+
+            VoteConsensusReached::<T>::insert(&consensus_key, true);
+
+            // Use helper function for consensus check
+            let (consensus_reached, positive_votes, required_for_consensus) =
+                Self::check_consensus_reached(instance, &root_id)?;
+
+            // If we didn't reach consensus through positive votes, notify admin
+            if !consensus_reached {
+                Self::deposit_event(Event::SummaryAcceptedWithoutConsensus {
+                    summary_instance: instance,
+                    root_id: root_id.clone(),
+                    positive_votes,
+                    required_votes: required_for_consensus,
+                });
+            }
+
+            Self::deposit_event(Event::WatchtowerConsensusReached {
+                summary_instance: instance,
+                root_id: root_id.clone(),
+                consensus_result: VotingStatus::Accepted,
+            });
```

Comment:
so we always reach consensus regaldless if we reached concensus? shouldn't we have something different if consensus_reached==false?